rfcs/text/0000-create-file-elf.md

```diff
@@ -0,0 +1,306 @@
+# 0000: Create the ELF sub-field of the File fieldset
+
+- Stage: **0 (strawperson)**
+- Date: **TBD**
+
+Create the Executable Linkable Format (ELF) sub-field, of the `file` top-level fieldset. This document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create the ELF sub-field within the `file.` fieldset. This will include 25 sub-fields.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.elf.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.elf.exports.name | keyword | Name of exported symbol |
+| file.elf.exports.type | keyword | Type of exported symbol |
+| file.elf.segments | keyword | ELF object segment list. |
+| file.elf.header.class | keyword | Header class of the ELF file. |
+| file.elf.header.data | keyword | Data table of the ELF header. |
+| file.elf.header.machine | keyword | Machine architecture of the ELF header. |
+| file.elf.header.os_abi | keyword | NEED TO ADD |
+| file.elf.header.type | keyword | Header type of the ELF file. |
+| file.elf.header.version | keyword | Version of the ELF header. |
+| file.elf.header.abi_version | keyword | Version of the ELF Application Binary Interface (ABI). |
+| file.elf.header.entrypoint | long | Header entrypoint of the ELF file. |
+| file.elf.header.object_version | keyword | "0x1" for original ELF files. |
+| file.elf.imports.name | keyword | Name of imported symbol |
+| file.elf.imports.type | keyword | Type of imported symbol |
+| file.elf.number_program_headers | long | Number of ELF Program Headers. |
+| file.elf.sections.flags | keyword | ELF Section List flags. |
```

Comment:
Can you make sure that the parent of these is actually documented too? This looks like `file.elf.sections` is intended to be `nested` to me.

Reply:
Oversight. Fix't.